File: Data Structures & Algorithms/palindrome-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        temp = []

        curr = head

        while curr:
            temp.append(curr.val)
            curr = curr.next
        
        l, r = 0, len(temp)- 1

        while l < r:
            if temp[l] != temp[r]:
                return False
            l,r = l+1, r-1
        
        return True
        

File: Data Structures & Algorithms/intersection-of-two-linked-lists/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        ListNode currA = headA;
        while (currA != null){
            ListNode currB = headB;
            while(currB != null){
                if (currA == currB){
                    return currA;
                }
                currB = currB.next;
            }
            currA = currA.next;
        }
        return null;
    }
}